src/App.js

```diff
@@ -240,35 +232,35 @@ class App extends React.Component {
       daoAddress,
       appsLoading,
       permissionsLoading,
-      resolveEntity,
-      resolveRole,
-      acl,
     } = this.state
     const { mode } = locator
     if (!mode) return null
 
     return (
       <AragonApp publicUrl="/aragon-ui/">
-        <Wrapper
-          historyBack={this.historyBack}
-          historyPush={this.historyPush}
-          locator={locator}
+        <PermissionsProvider
           wrapper={wrapper}
           apps={apps}
-          appsLoading={appsLoading}
           permissions={permissions}
-          permissionsLoading={permissionsLoading}
-          account={account}
-          walletNetwork={walletNetwork}
-          walletWeb3={walletWeb3}
-          web3={web3}
-          daoAddress={daoAddress}
-          transactionBag={transactionBag}
-          connected={connected}
-          resolveEntity={resolveEntity}
-          resolveRole={resolveRole}
-          acl={acl}
-        />
+        >
+          <Wrapper
+            historyBack={this.historyBack}
+            historyPush={this.historyPush}
+            locator={locator}
+            wrapper={wrapper}
+            apps={apps}
+            appsLoading={appsLoading}
+            permissions={permissions}
+            permissionsLoading={permissionsLoading}
+            account={account}
+            walletNetwork={walletNetwork}
+            walletWeb3={walletWeb3}
+            web3={web3}
+            daoAddress={daoAddress}
+            transactionBag={transactionBag}
+            connected={connected}
+          />
+        </PermissionsProvider>
```

Comment:
Thoughts on moving the `PermissionsProvider` down into the `Wrapper` instead? Not sure what the best practice with top level context is, but since only the `Wrapper` needs permissions info it might make sense.

Otherwise, I think we can remove a bunch of props from `Wrapper` too :).

Reply:
My thinking was that if we add more context providers, they could all be declared at the `App` level, so that `App` would be responsible of initializing `aragonjs-wrapper` and other data, that would then be passed to the contexts.

That way, `Wrapper` and `Onboarding` could focus more on the UI only, and `App` on the initialization. Also, if we convert `aragonjs-wrapper` to one or several contexts, `App` would only pass a few props to `Wrapper` and `Onboarding`. What do you think?

Reply:
That sounds great!